extract api to configure aspnet hosting to expose .NET Interactive api (#802)
dotnet/interactive: docs/javascript-overview.md

* `interactive.csharp`
* `interactive.fsharp`
* `interactive.pwsh`
* `interactive.value`

## Loading external JavaScript modules at runtime

Sometimes you might need to import JavaScript modules in your notebook. You can use the `interactive` object to do so.

Here is an example.

> ![image](https://user-images.githubusercontent.com/375556/95768858-e6be3500-0cae-11eb-925c-53d46670bd09.png)

Reviewer: Let's use code rather than images to document APIs. This will be hard to search for otherwise.

Author: removed image and made test example that users can actually use and try

> The `interactive.configureRequire` takes as input a [requirejs configuration object](https://www.tutorialspoint.com/requirejs/requirejs_configuration.htm) and returns a function that you can now use to load the dependecies.

Reviewer: Let's add a simple, working code sample.

Author: done

Sometimes you might need to import JavaScript modules in your notebook. You can use the `interactive` object to do so.

> Here is an example that configures requirejs to load d3.js from cdn.

Reviewer: ```suggestion
Here is an example that configures RequireJS to load D3.js from a CDN.
```

```js
#!js
dtreeRequire = interactive.configureRequire({
    paths: {
        d3: "https://d3js.org/d3.v5.min"
    }
});
```

> Now you can require the module and use it. Using `#!html` to inject an `svg` element and then `#!js` to load the `d3.js` library and append a circle to the dom

Reviewer: ```suggestion
Now you can require the module and use it. You can use the `#!html` magic command to inject an SVG element and then the `#!js` magic command to load the D3.js library and append a circle.
```

```
#!html

> <sgv id"renderTarget"></svg>

Reviewer: ```suggestion
<svg id"renderTarget"></svg>
```

Author: fixed

#!js
#!html

> <svg id="renderTarget" width=300 height=300></svg>

Reviewer: You have an extra `#!js` here and the `renderTarget` is being added twice.

Author: fixed
